packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2680,6 +2680,16 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           return super.toAssignable(this.typeCastToParameter(node), isLHS);
         case "TSParameterProperty":
           return super.toAssignable(node, isLHS);
+        case "ParenthesizedExpression":
+          if (
+            node.expression.type === "TSAsExpression" ||
+            node.expression.type === "ParenthesizedExpression"
```

Comment:
I guess `TSNonNullExpression` and `TSTypeAssertion` should be here too

Reply:
Yes, I tried running some tests on those, but they work like magic😁